erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py

```diff
@@ -179,3 +203,14 @@ def update_asset(self, asset_value=None):
 		)
 		asset.flags.ignore_validate_update_after_submit = True
 		asset.save()
+
+
+@frappe.whitelist()
+def value_of_accounting_dimension(asset_name):
+	acc_dimension_fields_value = {}
+	fields = frappe.get_list("Accounting Dimension", pluck="fieldname")
+	for field in fields:
+		value = frappe.db.get_value("Asset", asset_name, field)
+		acc_dimension_fields_value[field] = value
+	acc_dimension_fields_value["cost_center"] = frappe.db.get_value("Asset", asset_name, "cost_center")
+	return acc_dimension_fields_value
```

Comment:
Avoid `get_value` inside a for loop, here you can fetch all the dimensions in a single query.

You can write this function like below:

```
dimension_fields = frappe.get_list("Accounting Dimension", pluck="fieldname") + ['cost_center']
return frappe.db.get_value("Asset", asset_name, fieldname=dimension_fields, as_dict=True)
```